src/core/lib/iomgr/ev_epoll1_linux.cc

```diff
@@ -383,6 +383,10 @@ static void fd_shutdown_internal(grpc_fd* fd, grpc_error* why,
   if (fd->read_closure->SetShutdown(GRPC_ERROR_REF(why))) {
     if (!releasing_fd) {
       shutdown(fd->fd, SHUT_RDWR);
+    } else {
+      if (epoll_ctl(g_epoll_set.epfd, EPOLL_CTL_DEL, fd->fd, nullptr) != 0) {
```

Comment:
According to the Linux manual page http://man7.org/linux/man-pages/man2/epoll_ctl.2.html, the _event_ argument cannot be null for **EPOLL_CTL_DEL** in kernel versions before 2.6.9. Do we need to be portable to Linux kernels before 2.6.9?

Also, does any existing application expect that the released fd is still within the epoll set?

Reply:
Done. Global presubmit tests look fine too. PTAL